code/modules/cm_preds/yaut_weapons.dm

```diff
@@ -117,6 +117,209 @@
+				if(has_chain) // Generating the chain for the effect, its a bulelt so it looks like youre "throwing it / it looks like its travelling"
+					var/obj/projectile/hook_projectile = new /obj/projectile(user.loc, create_cause_data("hook"), user)
+					var/datum/ammo/ammoDatum = GLOB.ammo_list[/datum/ammo/yautja/gauntlet_hook]
+					hook_projectile.generate_bullet(ammoDatum, bullet_generator = user)
+					hook_projectile.bound_beam = hook_projectile.beam(user, "chain", 'icons/effects/beam.dmi', 3 SECONDS)
+					hook_projectile.fire_at(target, user, user, ammoDatum.max_range, ammoDatum.shell_speed)
+					if(prob(1))
+						var/list/heard = get_mobs_in_view(7, user)
+						for(var/mob/mob_in_range in heard)
+							if(mob_in_range.ear_deaf)
+								heard -= mob_in_range
+						user.langchat_speech(chain_message, heard, GLOB.all_languages, message_color, TRUE)
+					addtimer(CALLBACK(target, TYPE_PROC_REF(/mob/living, throw_carbon), target, reverse_facing, 5, SPEED_VERY_FAST), 0.5 SECONDS)
+					user.spin_circle()
+				user.throw_carbon(target, facing, punch_knockback, SPEED_VERY_FAST,)
+				target.visible_message(SPAN_XENOHIGHDANGER("[user] hits [target] with an extremely strong punch, sending them flying!"))
+				combo_counter = 0
+			user.flick_attack_overlay(target, "slam")
+			playsound(target, sound_to_play, 50, 1)
+
+		if(INTENT_GRAB)
+			if(!(HAS_TRAIT(target, TRAIT_KNOCKEDOUT) || target.stat == UNCONSCIOUS))
+				return
+
+			if(!do_after(user, 0.8 SECONDS, INTERRUPT_ALL, BUSY_ICON_HOSTILE, numticks = 2))
+				return
+
+			if(executing)
+				return
+
+			if(!executing)
+				executing = TRUE
+				user.visible_message(SPAN_XENOHIGHDANGER("[user] grabs [target] and slowly lifts them above their head before smashing them down!"))
+				playsound(target, 'sound/effects/bone_break1.ogg', 50, 1)
+				playsound(user, 'sound/voice/pred_roar5.ogg', 50, 1)
+				target.apply_damage(60, ARMOR_MELEE, BRUTE, "chest", 5)
+				target.death(create_cause_data("back broken", user), FALSE)
+				animate(target, pixel_y = target.pixel_y + 64, time = 4, easing = SINE_EASING)
+				sleep(4)
+				playsound(target, 'sound/effects/bang.ogg', 25, 0)
+				playsound(target,"slam", 50, 1)
+				animate(target, pixel_y = 0, time = 4, easing = BOUNCE_EASING)
+				sleep(10) // This is so people dont spam click and lineup 300 executions, i know its probably not the best way to do it.
+				executing = FALSE
+
+		if((INTENT_HARM)) // This is how you farm combo counters, so there's no special interaction.
+			playsound(target, sound_to_play, 50, 1)
+			user.flick_attack_overlay(target, "slam")
+
+
+	if(target != user)
+		if(target.stat == DEAD)
+			return
+		combo_counter++
+		COOLDOWN_START(src, combo_timeout, 15 SECONDS) // This just sets combo to 0 so people dont "pre-stack combos"
+		START_PROCESSING(SSobj, src)
+
+
+/obj/item/weapon/bracer_attachment/chain_gauntlets/process()
+	if(COOLDOWN_FINISHED(src, combo_timeout))
+		combo_counter = 0
+		STOP_PROCESSING(SSobj, src)
```

Comment:
could you avoid processing the obj by checking for whether the cooldown has finished at the start of every attack (and early returning with a normal punch if it expired)